py/serve.py

```diff
@@ -798,27 +807,34 @@ def issue_access_token(user_id):
         return response
 
 
-def make_user(name=None):
+def make_user():
     """Writes a new user ID and returns it.
 
-    The "name" is reserved for GitHub login values which uniquely identify
-    GitHub users.
-
-    Args:
-        name: Hopefully a readable label for the user, not currently used in UI.
-
     Returns:
         The 32-character generated UUID of the user, currently used in the UI.
     """
     with flask.g.db.cursor() as cursor:
         query = psycopg2.sql.SQL('INSERT INTO users VALUES (%s, %s, %s)')
         user_id = uuid.uuid4().hex
         timestamp = int(time.time())
-        cursor.execute(query, [user_id, name, timestamp])
+        cursor.execute(query, [user_id, None, timestamp])
         flask.g.db.commit()
     return user_id
 
 
+def migrate_user(name, user_id):
```

Comment:
I would suggest we check to see if the name is already set in this function and raise an error that the account is already associated with a GitHub `name`. Otherwise, it seems like I could hijack the data in your account by setting my access token to match yours, then logging in to my personal GitHub and transferring ownership.

Reply:
Note that `migrate_user` is only called if `SELECT user_id FROM users WHERE name=%s` fails. So in the situation where I hijacked your access token and then logged in to GitHub, the access token would be overridden by the call to `issue_access_token` at the end of `github_callback` and `migrate_user` would never be called.

Added some comments to help clarify.

Reply:
Actually, I think I'm wrong about this. If you didn't already have a GitHub account registered that query would fail. I'll update.

Reply:
@connorcoley PTAL